src/ir/term/mod.rs

```diff
@@ -1664,6 +1669,28 @@ fn eval_value(vs: &mut TermMap<Value>, h: &FxHashMap<String, Value>, c: Term) ->
             }
             Value::Array(res)
         }
+        Op::Rot(i) => {
+            let a = vs.get(&c.cs[0]).unwrap().as_array().clone();
+            let iter = match check(&c.cs[0]) {
+                Sort::Array(k, _, s) => (*k).clone().elems_iter_values().take(s).enumerate(),
+                _ => panic!("Input type should be Array"),
+            };
+            let (mut res, len) = match check(&c) {
+                Sort::Array(k, v, n) => (Array::default((*k).clone(), &v, n), n),
+                _ => panic!("Output type of rot should be Array"),
+            };
```

Comment:
Can you get this from the input type?

Reply:
By input type, do you mean the first argument instead of `c`?

Reply:
Yes. i.e., in the above match.